game/plugins/src/main/kotlin/gg/rsmod/plugins/content/skills/thieving/pickpocketing/PickpocketInfo.kt

```diff
@@ -0,0 +1,461 @@
+package gg.rsmod.plugins.content.skills.thieving.pickpocketing
+
+import gg.rsmod.game.model.weight.impl.WeightItem
+import gg.rsmod.game.model.weight.impl.WeightItemSet
+import gg.rsmod.plugins.api.cfg.Items
+import gg.rsmod.plugins.api.cfg.Npcs
+
+/**
+ * @ids = an array of NPC ids for them to pickpocket
+ * @exp = the amount of experience given per pickpocket
+ * @lvl = the level requirement to pickpocket that npc
+ * @npcName = the name of the NPC for the chat messages
+ * @rewards = a weighted set of possible item rewards
```

Comment:
I don't understand the benefit of this if you could clarify. If I were to make them an Array, that just adds an extra level of conversion that isn't needed, no?

Reply:
There's several benefits to taking this approach, including:

- Remove boilerplate code (``add`` on every single ``WeightItem``)
- Less error-prone
- If we ever want to change the rewards to __not__ use ``WeightItemSet`` you only have to remove it in a single place instead of every entry